src/main/scala/goatrodeo/util/ArtifactWrapper.scala

```diff
@@ -68,19 +118,180 @@ case class ByteWrapper(bytes: Array[Byte], fileName: String)
 
   override def listFiles(): Vector[ArtifactWrapper] = Vector()
 
-  override def getParentDirectory(): File = File("/")
+  override def getParentDirectoryPath(): String = "/"
 
   override def getCanonicalPath(): String = "/"
 
-  // override def asFile(): (File, Boolean) =
-  //   (Helpers.tempFileFromStream(asStream(), true, ".goat"), true)
-
   override def isDirectory(): Boolean = false
 
   override def asStream(): InputStream = ByteArrayInputStream(bytes)
 
   override def name(): String = fileName
 
   override def size(): Long = bytes.length
+
+  override def fileExtension(): Option[String] = {
+    logger.debug(s"File Extension against ByteWrapper, returning None")
+    None
+  }
+}
+
+case class ISOFileWrapper(f: File, isoReader: IsoFileReader, deleteOnFinalize: Boolean)
+  extends ArtifactWrapper {
+
+  private val logger = Logger("ISOFileWrapper")
+
+
+  override protected def finalize(): Unit = {
+    if (deleteOnFinalize) {
+      f.delete()
+    }
+  }
+
+  /**
+   * ISOFileWrapper is an "outer" wrapper for the ISO file itself; all the actual internal files
+   * are of type `InternalISOFileWrapper`; here for now we'll return an empty stream
+   * TODO - figure out better pathing
+   */
+  override def asStream(): InputStream =
```

Comment:
This should not return an empty stream. It should return the stream for the file itself. This is how the gitoid for the file is computed.

Put another way, all the gitoids for all the ISOs will be the same, which is a bad thing.